src/Juvix/Compiler/Core/Translation/FromInternal.hs

```diff
@@ -423,6 +425,20 @@ goAxiomDef a = case a ^. Internal.axiomBuiltin >>= builtinBody of
           )
       Internal.BuiltinString -> Nothing
       Internal.BuiltinIO -> Nothing
+      Internal.BuiltinTrace ->
+        Just
+          ( mkLambda'
+              ( mkLambda'
+                  ( mkLambda'
+                      ( mkLambda'
+                          ( mkBuiltinApp' OpTrace [mkVar' 1, mkVar' 0]
```

Comment:
Why 4 lambda here? I don't understand this.

Besides, translating `trace` this way is bad for evaluation order. Then `trace 3 (trace 4 unit)` will print "4\n3\n" instead of "3\n\4". I think it's better to translate it the way `if` is translated, with the arguments immediately supplied to `mkBuiltinApp'` -- then the laziness of `trace` with respect to the second argument is preserved.

Reply:
You need for lambdas, because the type I have in mind for trace is {A : Type} -> {B : Type} -> A -> B -> B

Reply:
Yes, sorry, I realised this about the type arguments.

But `trace` should be lazy.